diff --git a/gdb/build-id.c b/gdb/build-id.c
index be74127..0bf46d5 100644
--- a/gdb/build-id.c
+++ b/gdb/build-id.c
@@ -1252,12 +1252,12 @@ debug_print_missing (const char *binary, const char *debug)
 			    _("Missing separate debuginfo for %s\n"), binary);
         if (debug != NULL)
 	  fprintf_unfiltered (gdb_stdlog, _("Try: %s %s\n"),
-#ifdef DNF_DEBUGINFO_INSTALL
-			      "dnf"
-#else
-			      "yum"
-#endif
-			      " --enablerepo='*debug*' install", debug);
+			      "dnf --enablerepo='*debug*' install", debug);
+
+        if (debug != NULL)
+	  fprintf_unfiltered (gdb_stdlog, _("Or try: %s %s\n"),
+			      "urpmi ", debug);
+
       }
 }
 
