diff -up gdb-17.1/gdb/python/lib/gdb/command/rpm-suggestions.py.omv~ gdb-17.1/gdb/python/lib/gdb/command/rpm-suggestions.py
--- gdb-17.1/gdb/python/lib/gdb/command/rpm-suggestions.py.omv~	2026-02-08 03:28:25.200711942 +0100
+++ gdb-17.1/gdb/python/lib/gdb/command/rpm-suggestions.py	2026-02-08 03:29:34.302662469 +0100
@@ -471,7 +471,6 @@ class rpm_suggestion_enabled(gdb.Paramet
 
     def __init__(self):
         super().__init__("rpm-suggestion enabled", gdb.COMMAND_NONE, gdb.PARAM_BOOLEAN)
-        self.value = True
 
 # The 'set/show rpm-suggestion enabled' command.
 class rpm_suggestion_build_id_mode(gdb.Parameter):
@@ -525,3 +524,12 @@ param_rpm_suggestion_build_id_mode = rpm
 
 # Create the 'info rpm-suggestions' commands.
 rpm_suggestion_info()
+
+# Enable the plugin by default
+def rpm_suggestion_finalize_setup():
+    try:
+        gdb.execute("set rpm-suggestion enabled on")
+    except:
+        pass
+
+gdb.post_event(rpm_suggestion_finalize_setup)
