diff -up gdb-17.1/gdb/python/py-objfile.c.omv~ gdb-17.1/gdb/python/py-objfile.c
diff -up gdb-17.1/gdb/python/python.c.omv~ gdb-17.1/gdb/python/python.c
--- gdb-17.1/gdb/python/python.c.omv~	2026-02-08 03:56:44.583217731 +0100
+++ gdb-17.1/gdb/python/python.c	2026-02-08 04:01:34.060807620 +0100
@@ -1894,14 +1894,14 @@ gdbpy_handle_missing_debuginfo (const st
     }
 
   /* Parse the result, and convert it back to the C++ object.  */
-  if (pyo_execute_ret == Py_None)
+  if (Py_IsNone (pyo_execute_ret.get ()))
     return {};
 
   if (PyBool_Check (pyo_execute_ret.get ()))
     {
       /* We know the value is a bool, so it must be either Py_True or
 	 Py_False.  Anything else would not get past the above check.  */
-      bool try_again = pyo_execute_ret.get () == Py_True;
+      bool try_again = PyObject_IsTrue (pyo_execute_ret.get ());
       return ext_lang_missing_file_result (try_again);
     }
 
