diff -up gdb-17.1/gdb/contrib/gdb-add-index.sh.57~ gdb-17.1/gdb/contrib/gdb-add-index.sh
--- gdb-17.1/gdb/contrib/gdb-add-index.sh.57~	2025-09-06 23:52:48.000000000 +0200
+++ gdb-17.1/gdb/contrib/gdb-add-index.sh	2025-12-21 18:17:47.905446085 +0100
@@ -25,6 +25,20 @@ READELF=${READELF:=readelf}
 PKGVERSION="@PKGVERSION@"
 VERSION="@VERSION@"
 
+GDB2=/usr/libexec/gdb
+if test -x $GDB2 && ! which $GDB &>/dev/null; then
+    GDB=$GDB2
+fi
+
+# We default to using /usr/bin/gdb.minimal if it's present.  See
+# https://bugzilla.redhat.com/show_bug.cgi?id=1695015 and
+# https://fedoraproject.org/wiki/Changes/Minimal_GDB_in_buildroot for
+# explanations.
+GDB3=/usr/bin/gdb.minimal
+if test -x $GDB3; then
+    GDB=$GDB3
+fi
+
 myname="${0##*/}"
 
 print_usage() {
