diff -p -up gdb-7.0/sim/Makefile.in.orig gdb-7.0/sim/Makefile.in
--- gdb-7.0/sim/Makefile.in.orig	2009-07-30 19:53:18.000000000 -0300
+++ gdb-7.0/sim/Makefile.in	2009-12-18 13:52:35.449269697 -0200
@@ -85,6 +85,8 @@ FLAGS_TO_PASS = \
 	"exec_prefix=$(exec_prefix)" \
 	"bindir=$(bindir)" \
 	"mandir=$(mandir)" \
+	"infodir=$(infodir)" \
+	"includedir=$(includedir)" \
 	"libdir=$(libdir)" \
 	"against=$(against)" \
 	"AR=$(AR)" \
