--- gdb-8.0.1/gdb/python/python.c.omv~	2018-02-14 21:07:22.559712601 +0100
+++ gdb-8.0.1/gdb/python/python.c	2018-02-14 21:15:41.816374526 +0100
@@ -1724,7 +1724,7 @@ do_start_initialization ()
 #ifdef IS_PY3K
   gdb_module = PyModule_Create (&python_GdbModuleDef);
   /* Add _gdb module to the list of known built-in modules.  */
-  _PyImport_FixupBuiltin (gdb_module, "_gdb");
+  _PyImport_FixupBuiltin (gdb_module, "_gdb", PyThreadState_Get()->interp->modules);
 #else
   gdb_module = Py_InitModule ("_gdb", python_GdbMethods);
 #endif
